src/ck3/data/religions.rs

```diff
@@ -69,7 +71,44 @@ impl DbKind for Religion {
         vd.field_item("family", Item::ReligionFamily);
 
         vd.req_field("doctrine");
-        vd.multi_field_item("doctrine", Item::Doctrine);
+        let mut categories: FnvHashMap<&str, Vec<Token>> = FnvHashMap::default();
+        vd.multi_field_validated_value("doctrine", |_, mut vd| {
+            vd.item(Item::Doctrine);
+            if let Some(category) = data.doctrines.category(vd.value().as_str()) {
+                let doctrine = vd.value();
+                if let Some(seen) = categories.get_mut(category.as_str()) {
+                    let picks = data.doctrines.number_of_picks(category.as_str());
+                    #[allow(clippy::cast_possible_wrap)]
+                    if picks == 1 {
+                        // SAFETY: we never push empty vecs into this hash
+                        let other_doctrine = &seen[0];
+                        if doctrine == other_doctrine {
+                            let msg = format!("duplicate doctrine {doctrine}");
+                            warn(ErrorKey::DuplicateField)
+                                .msg(msg)
+                                .loc_msg(doctrine, "doctrine")
+                                .loc(other_doctrine, "earlier doctrine")
+                                .push();
+                        } else {
+                            let msg =
+                                format!("{doctrine} and {other_doctrine} are both from {category}");
```

Comment:
this can be a bit ambiguous. best to mention that you can only pick one doctrine from this category

Reply:
Hmm I didn't do that because it's the default. There's basically only one category that gets number_of_picks and that's the core tenets. So picks > 1 is the exception

Reply:
Oh, I could put that in the info though